### PR TITLE
Modal Dialogs open under Modeless Dialogs

### PR DESCRIPTION
**Pull request by NancyMcCallB**
Update the way ZIndex is managed for dialogs such that Modal dialogs do not open under existing modeless dialogs.

### CODE REVIEW

**Review by pmconne — APPROVED** [review]
@NancyMcCallB can you update CODEOWNERS to add itwinjs-core-ui as owners of the new ui test apps like appui-test-app? That way you won't need admin review when modifying them.

File: ui/appui-react/src/appui-react/dialog/ModelessDialogManager.tsx [hunk 1]
```diff
@@ -96,7 +79,7 @@ export class ModelessDialogManager {
         ModelessDialogManager._idArray.splice(index, 1);
 
       if (ModelessDialogManager.activeDialog === undefined)
-        ModelessDialogManager._topZIndex = ModelessDialogManager.getDialogZIndexDefault();
+        DialogManagerBase.topZIndex = DialogManagerBase.getDialogZIndexDefault();
```
Comment (raplemie) [inline ui/appui-react/src/appui-react/dialog/ModelessDialogManager.tsx:82]: If there is no more ModelessDialog active, we reset the `topZIndex`, however, there could still be normal dialog open, which will still want to use the previous `topZIndex`. (same thing in the other way)

**Review by raplemie — COMMENTED** [review]
There might be an issue when closing dialog where we reset the top but other type of dialog may still be open.

File: ui/appui-react/src/appui-react/dialog/ModelessDialogManager.tsx (hunk 1 above)
Comment (NancyMcCallB) [reply ui/appui-react/src/appui-react/dialog/ModelessDialogManager.tsx:82]: Right. All the dialogs will be the default zIndex, unless multiple dialogs are open and they get reordered. The reordering is currently broken, so I'll look at that problem when I fix the reodering.

Probably, we want to move the whole thing down to the DialogManagerBase class.

**Review by raplemie — APPROVED** [review]
LGTM!

**Pull request merged**